addons/mail/static/src/utils/common/format.js

```diff
@@ -208,6 +209,39 @@ function generateMentionsLinks(body, { partners = [], threads = [], specialMenti
     return body;
 }
 
+/**
+ * @param body {string}
+ * @param messages {Array}
+ * @return {string}
+ */
+function generateMessageLinks(body, messages = []) {
+    for (const message of messages) {
+        const { model, id: threadId } = message.thread;
+        const url = generateMessageUrl(message);
+        const regex = new RegExp(`\\b${url}\\b`, "g");
+        const href = `href='${url}'`;
+        const className =
+            model === "discuss.channel" ? "o_channel_message_redirect" : "o_mail_message_redirect";
+        const attClass = `class='${className}'`;
+        const dataId = `data-id='${message.id}'`;
+        const dataOeId = `data-oe-id='${threadId}'`;
+        const dataOeModel = `data-oe-model='${model}'`;
+        const target = "target='_blank'";
+        const link = `<a ${href} ${attClass} ${dataId} ${dataOeId} ${dataOeModel} ${target} contenteditable="false">${url}</a>`;
```

Comment:
The content of the link should be wrapped inside `_escapeEntities` (see linkify function)